docs/reference/remap.cue

```diff
@@ -65,4 +64,205 @@ remap: {
 		}
 	}
+
+	// TRL type system
+	types: [TypeName=string]: {
+		#Use: "parameter" | "return"
+
+		description: string
+		use: [#Use, ...#Use]
+	}
+
+	types: {
+		"any": {
+			description: "A stand-in for any type."
+			use: ["parameter"]
+		}
+		"array": {
+			description: "A list of items."
+			use: ["parameter", "return"]
+		}
+		"boolean": {
+			description: "`true` or `false`."
+			use: ["parameter", "return"]
+		}
+		"float": {
+			description: "A 64-bit floating-point number."
+			use: ["parameter", "return"]
+		}
+		"map": {
+			description: """
+				A key-value map in which keys are strings and values can be of any TRL type,
+				including maps.
+				"""
+			use: ["parameter", "return"]
+		}
+		"integer": {
+			description: "A 64-bit integer."
+			use: ["parameter", "return"]
+		}
+		"null": {
+			description: "No value."
+			use: ["return"]
+		}
+		"path": {
+			description: "An event field."
+			use: ["parameter"]
+		}
+		"regex": {
+			description: "A regular expression."
+			use: ["parameter"]
+		}
+		"string": {
+			description: """
+				A sequence of characters. Remap converts strings in scripts to [UTF-8](\(urls.utf8))
+				and replaces any invalid sequences with `U+FFFD REPLACEMENT CHARACTER` (�).
+				"""
+			use: ["parameter", "return"]
+		}
+		"timestamp": {
+			description: "A string formatted as a timestamp."
+			use: ["parameter", "return"]
+		}
+	}
+
+	// TRL syntax
+	#Operators: [_category=string]: [_op=string]: string
+
+	syntax: [RuleName=string]: {
+		name:        RuleName
+		href:        string // Ensures that we don't end up with clashing anchors
+		description: string
+		examples: [string, ...string]
+		operators?: #Operators
+	}
+
+	syntax: {
+		"Dot notation": {
+			href: "dot-notation"
+
+			description: """
+				In TRL, a dot (`.`) holds state across the script. At the beginning of the script,
+				it represents the event arriving into the transform. Take this JSON event data as an
+				example:
+
+				```json
+				{"status_code":200,"username":"booper1234","message":"Successful transaction"}
+				```
+
+				In this case, the event, represented by the dot, has three fields: `.status_code`,
+				`.username`, and `.message`. You can assign new values to the existing fields
+				(`.message = "something different"`), add new fields (`.new_field = "new value"`),
+				delete fields (`del(.username)`), store those values in variables (`$code =
+				.status_code`), and more.
+
+				The dot can also represent nested values, for example `.transaction.id` or
+				`.geo.latitude`.
+				"""
+			examples: [
+				".",
+				".status_code",
+				".message",
+				".username",
+				".transaction.id",
+				".geo.latitude",
+			]
+		}
+
+		"Functions": {
+			href: "functions"
+
+			description: """
+				In TRL, functions act just like they do in standard programming languages. They can
+				take inputs (or no input) and either return values or make an
+				[assertion](#assertion) using the [`assert`](#assert) function.
```

Comment:
I'm not convinced it makes sense to call out `assert` specifically.

Functions either return a value, or an error (`assert` returns `null` when it passes, and an error when it doesn't).